ncm-network/src/main/perl/nmstate.pm

```diff
@@ -0,0 +1,720 @@
+If this fails, the component reverts all settings to the previous values. This is no different to network module.
+
+During this test, a sleep value of 15 seconds is used to make sure the restarted network
+is fully restarted (routing may need some time to come up completely).
+
+Because of this, configuration changes may cause the ncm-ncd run to take longer than usual.
+
+Be aware that configuration changes can also lead to a brief network interruption.
+=cut
+
+use parent qw(NCM::Component::network);
+use NCM::Component::network;  # required for the import of the (default) exports
+
+our $EC = LC::Exception::Context->new->will_store_all;
+use EDG::WP4::CCM::TextRender;
+use Readonly;
+
+Readonly my $NMSTATECTL => '/usr/bin/nmstatectl';
+Readonly my $NMCLI_CMD => '/usr/bin/nmcli';
+# pick a config name for nmstate yml to configure dns-resolver: settings. if nm_manage_dns=true
+Readonly my $NM_RESOLV_YML => "/etc/nmstate/resolv.yml";
+Readonly my $NM_DROPIN_CFG_FILE => "/etc/NetworkManager/conf.d/90-quattor.conf";
+
+# generate the correct fake yaml boolean value so TextRender can convert it in a yaml boolean
+Readonly my $YTRUE => $EDG::WP4::CCM::TextRender::ELEMENT_CONVERT{yaml_boolean}->(1);
+Readonly my $YFALSE => $EDG::WP4::CCM::TextRender::ELEMENT_CONVERT{yaml_boolean}->(0);
+
+use constant IFCFG_DIR => "/etc/nmstate";
+
+sub iface_filename
+{
+    my ($self, $iface) = @_;
+    return $self->IFCFG_DIR . "/$iface.yml";
+}
+
+# Given the configuration in nmconnection
+# Determine if this is a valid interface for ncm-network to manage,
+# Return arrayref tuple [interface name, ifdown/ifup name] when valid,
+# undef otherwise.
+sub is_valid_interface
+{
+    my ($self, $filename) = @_;
+
+    # Very primitive, based on regex only
+    # matchs eth0.yml bond0.yml, or bond0.101.yml
+    if (
+        $filename =~ m{
+            # Filename is either right at the beginning or following a slash
+            (?: \A | / )
+            # $1 will capture for example:
+            # eth0  bond1  eth0.101  bond0.102
+            ( \w+ \d+ (?: \. \d+ )? )
+            # Suffix (not captured)
+            \. yml \z
+        }x
+    ) {
+        # name and id for nmstate, this will make connection id and name the same.
+        my $name = $1;
+        # previous is_valid_interface reuqires suffix, not concerned about this in nmstate so just return $name as suffix too.
+        return [$name, $name];
+    } else {
+        return;
+    };
+}
+
+# By default, NetworkManager on Red Hat Enterprise Linux (RHEL) 8+ dynamically updates the /etc/resolv.conf
+# file with the DNS settings from active NetworkManager connection profiles. we manage this using ncm-resolver.
+# so disable this unless nm_manage_dns = true. resolver details can be set using nmstate but not doing this now.
+sub disable_nm_manage_dns
+{
+    my ($self, $manage_dns, $nwsrv) = @_;
+    my @data = ('[main]');
+    
+    if ( $manage_dns ) {
+        # set nothing, will use default.
```

Comment:
if the behaviour of  `nm_manage_dns = true` means to not disable it, it needs a very clear explanaiton in the the schema.
i would force it here and not rely on default behaviour (eg another config file might set it to none).